diff --git a/lib/binding.ts b/lib/binding.ts
index f5963a02e39c344a4de19d682c1b7c1e20ba8528..9129e2a7219a228bb98f0fd2a745ca492d90b6ed 100644
--- a/lib/binding.ts
+++ b/lib/binding.ts
@@ -1,10 +1,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-import { NativeModules } from 'react-native';
+import { NativeModules, TurboModuleRegistry } from 'react-native';
 import type { OrtApi as OrtApiType } from './api';
 
-export const Module = NativeModules.Onnxruntime;
+export const Module = NativeModules.Onnxruntime ?? TurboModuleRegistry.get('Onnxruntime');
 
 declare global {
   var OrtApi: OrtApiType; // eslint-disable-line no-var
